File: le-domain/src/main/java/com/latticeengines/domain/exposed/dataplatform/HasProperty.java
package com.latticeengines.domain.exposed.dataplatform;

import java.util.Map.Entry;
import java.util.Set;

public interface HasProperty {

    Object getPropertyValue(String key);

    void setPropertyValue(String key, Object value);

    Set<Entry<String, Object>> getEntries();
}

File: le-sparkdb/src/main/scala/com/latticeengines/sparkdb/operator/HasProperty.scala
package com.latticeengines.sparkdb.operator

import java.util.HashMap
import java.util.Map
import java.util.Map.Entry
import java.util.Set

import com.latticeengines.domain.exposed.dataplatform.{HasProperty => DomainHasProperty}

trait HasProperty extends DomainHasProperty {

  private var map: Map[String, Any] = new HashMap[String, Any]()
  
  override def getPropertyValue(key: String): Object = map.get(key).asInstanceOf[Object]
  
  override def setPropertyValue(key: String, value: Any) = map.put(key, value)
  
  override def getEntries(): Set[Entry[String, Object]] = map.entrySet().asInstanceOf[Set[Entry[String, Object]]]

}